### PR TITLE
Add mode option to IIB builds

### PR DESCRIPTION
**Pull request by Allda**
The mode option allows users to configure how the update graph will be updated. The option is set in ci.yaml configuration file and is passed to the IIB build. The default value for the mode is "replaces".

JIRA: ISV-3837

### CODE REVIEW

File: ansible/roles/operator-pipeline/templates/openshift/pipelines/community-hosted-pipeline.yml [hunk 1]
```diff
@@ -251,6 +267,12 @@ spec:
         # Adding a commit hash as a suffix to the image tag
         - name: index-image-destination-tag-suffix
           value: "-$(params.git_commit)"
+        - name: upgrade-graph-mode
+          value: "$(tasks.read-config.results.upgrade-graph-mode)"
+        # Production IIB instance doesn't support the upgrade path feature yet.
+        # TODO: switch it to PROD when the feature is available
+        - name: iib_url
```
Comment (jedinym) [inline ansible/roles/operator-pipeline/templates/openshift/pipelines/community-hosted-pipeline.yml:274]: Should we create a Task ticket for this?

File: operator-pipeline-images/operatorcert/entrypoints/index.py [hunk 2]
```diff
@@ -113,15 +119,17 @@ def add_bundle_to_index(
     iib_url: str,
     indices: List[str],
     image_output: str,
+    mode: Optional[str] = None,
```
Comment (jedinym) [inline operator-pipeline-images/operatorcert/entrypoints/index.py:122]: Do we need a default parameter here if we always provide the option?

File: operator-pipeline-images/operatorcert/entrypoints/index.py [hunk 3]
```diff
@@ -53,6 +53,12 @@ def setup_argparser() -> argparse.ArgumentParser:  # pragma: no cover
         default="",
         help="A tag suffix to append to the index image when copying to the destination.",
     )
+    parser.add_argument(
+        "--mode",
+        choices=["replaces", "semver", "semver-skippatch"],
+        default="",
```
Comment (jedinym) [inline operator-pipeline-images/operatorcert/entrypoints/index.py:59]: I think having the default as `None` would be a bit cleaner, but your call.

**Review by jedinym — APPROVED** [review]

File: ansible/roles/operator-pipeline/templates/openshift/pipelines/community-hosted-pipeline.yml (hunk 1 above)
Comment (Allda) [reply ansible/roles/operator-pipeline/templates/openshift/pipelines/community-hosted-pipeline.yml:274]: I am in sync with a iib developer who takes care of that. I'll change it once it is ready.

File: operator-pipeline-images/operatorcert/entrypoints/index.py (hunk 2 above)
Comment (Allda) [reply operator-pipeline-images/operatorcert/entrypoints/index.py]: Fixed

File: operator-pipeline-images/operatorcert/entrypoints/index.py (hunk 3 above)
Comment (Allda) [reply operator-pipeline-images/operatorcert/entrypoints/index.py]: Fixed

**Pull request merged**